gpkit/constraints/set.py

```diff
@@ -183,22 +184,22 @@ def flat(self, constraintsets=True):
                     yield yielded_constraint
 
     def subinplace(self, subs):
-        "Substitutes in place."
+        """Substitutes in place, updating self.substitutions accordingly.
+
+        Keys substituted with `subinplace` are no longer present, so if such a
+        key is also in self.substitutions that substitution is now orphaned. If
+        `subs[key]` describes some key in the ConstraintSet (i.e. one key has
+        been substituted for another), then a substitution is added, mapping
+        the orphaned value to this new key; otherwise, an error is raised.
+        """
```

Comment:
Nope, we now (more cleanly) raise the error during `self.substitutions.parse_and_index(value)`